06. 자료구조(스택, 큐)/07. 교육과정설계/이서인/index.js

```diff
@@ -1,9 +1,20 @@
 function solution(need, plan) {
-  let answer = 'YES';
+  let answer = [];
+  let needIndex = 0;
+  const isNeedCourse = (el) => el === need[needIndex];
 
-  return answer;
+  [...plan].forEach((el) => {
+    if (isNeedCourse(el)) {
+      answer.push(needIndex);
+      needIndex++;
+    }
+  });
+  return answer.length === need.length &&
+    answer.every((val, index) => val === index)
+    ? "YES"
+    : "NO";
```

Comment:
다른 풀이를 보고 떠오른 조건에 맞는 것을 필터링해서 비교하는 방법입니다~

```suggestion
  const palnNeed = [...plan].filter((el) => {
    if (isNeedCourse(el)) {
      needIndex++;
      return true;
    }
    
    return false;
  }).join('');
  
  return palnNeed === need ? "YES" : "NO";
```

Reply:
👍🏻오 좋은데요?!??!?!?!